src/app/organizer/DropDown.tsx

```diff
@@ -25,30 +32,117 @@ function MenuItem({ item }: { item: DropDownItem }) {
   );
 }
 
+function Menu({ items }: { items: DropDownItem[] }) {
+  return (
+    <div className={styles.menu}>
+      {items.map((item) => (
+        <MenuItem key={item.id} item={item} />
+      ))}
+    </div>
+  );
+}
+
+function OrderedMenuItem({ item, index }: { item: DropDownItem; index: number }) {
+  const { checked, content, onItemSelect } = item;
+
+  return (
+    <Draggable draggableId={item.id} index={index}>
+      {(provided) => (
+        <div
+          className={`${styles.orderedMenuItem} ${item.checked ? '' : ' disabled'}`}
+          data-index={index}
+          ref={provided.innerRef}
+          {...provided.draggableProps}
+          onClick={onItemSelect}
+        >
+          <span className={styles.dragHandle} {...provided.dragHandleProps}>
+            <AppIcon icon={reorderIcon} className="reorder-handle" />
+          </span>
+          <span className={styles.name}>{content}</span>
+          <span className={styles.checkIcon}>
+            <AppIcon
+              icon={checked ? enabledIcon : unselectedCheckIcon}
+              className="checked-toggle-icon"
+            />
+          </span>
+        </div>
+      )}
+    </Draggable>
+  );
+}
+
+function OrderedMenu({
+  items,
+  onOrderChange
+}: {
+  items: DropDownItem[];
+  onOrderChange(items: DropDownItem[]): void;
+}) {
+  const onDragEnd = useCallback(
+    (result: DropResult) => {
+      // dropped outside the list
+      if (!result.destination) {
+        return;
+      }
+
+      const reorderedItems = [...items];
+      const [moved] = reorderedItems.splice(result.source.index, 1);
+      reorderedItems.splice(result.destination.index, 0, moved);
+
+      onOrderChange(reorderedItems);
+    },
+    [items, onOrderChange]
+  );
+
+  return (
+    <DragDropContext onDragEnd={onDragEnd}>
+      <Droppable droppableId="droppableMenu">
+        {(provided) => (
+          <div className={styles.orderedMenu} ref={provided.innerRef} {...provided.droppableProps}>
+            {items.map((item, index) => (
+              <OrderedMenuItem key={item.id} item={item} index={index} />
+            ))}
+          </div>
+        )}
+      </Droppable>
+    </DragDropContext>
+  );
+}
+
 function DropDown({
   buttonText,
   buttonDisabled,
-  dropDownItems
+  dropDownItems,
+  onOrderChange
 }: {
   buttonText: string;
   buttonDisabled?: boolean;
   dropDownItems: DropDownItem[];
+  onOrderChange?(items: DropDownItem[]): void;
 }) {
   const [dropdownOpen, setDropdownOpen] = useState(false);
+  const onClickOutside = useCallback(() => setDropdownOpen(false), [setDropdownOpen]);
+  const onButtonClick = useCallback(() => setDropdownOpen(!dropdownOpen), [
```

Comment:
If you wrap `DropDown` in `React.memo`, you probably don't need these `useCallback` hooks. That's because the component would only re-render if any of its props changed, and if any of the props changed then this will all re-render anyway.

In general you only need `useCallback` if you have a component that wants a stable referential identity for the callback function - for example, if you're passing that callback to another component that *does* use `React.memo`. In that case you'll avoid extra re-renders. But if the component you're calling doesn't use `React.memo` or `shouldComponentUpdate`, then it's going to re-render whenever this component renders anyway, so having a new function reference won't make any difference.